### PR TITLE
[NCL-3800] Unification of status work

### PR DESCRIPTION
**Pull request by thescouser89**
### Checklist:

* [ ] Have you added a note in the [CHANGELOG wiki](https://github.com/project-ncl/pnc/wiki/Changelog) for your change if user-facing?
* [ ] Have you added unit tests for your change?

### CODE REVIEW

File: constants/src/main/java/org/jboss/pnc/enums/BuildStatus.java [hunk 1]
```diff
@@ -55,51 +55,58 @@
...
-    UNKNOWN,
+    NEW;
 
-    /**
-     * There have not been any builds of this configuration, essentially the same meaning
-     * as a null status
-     */
-    NONE;
+    // isFinal means it's in the final state, it could also have failed
+    private final boolean isFinal;
```
Comment (jbartece) [inline constants/src/main/java/org/jboss/pnc/enums/BuildStatus.java]: Why do we need this boolean?

Comment (thescouser89) [reply constants/src/main/java/org/jboss/pnc/enums/BuildStatus.java]: To know if the build is in a final state. Aka if it's done (whether successful or not)

File: constants/src/main/java/org/jboss/pnc/enums/BuildStatus.java [hunk 2]
```diff
@@ -20,21 +20,21 @@
...
- * Status of a running or completed build.
+ * Status of a running or isFinal build.
  *
  * @author Jakub Bartecek &lt;jbartece@redhat.com&gt;
  *
  */
 public enum BuildStatus {
     /**
-     * Build completed successfully
+     * Build isFinal successfully
```
[the remark this answers is not in the document]
Comment (thescouser89) [reply constants/src/main/java/org/jboss/pnc/enums/BuildStatus.java]: That's intellij being too aggressive. Will fix it. Thanks for reporting it!

File: constants/src/main/java/org/jboss/pnc/enums/BuildStatus.java (hunk 1 above)
Comment (jbartece) [reply constants/src/main/java/org/jboss/pnc/enums/BuildStatus.java]: Do we have any use-case when we need it?

Comment (thescouser89) [reply constants/src/main/java/org/jboss/pnc/enums/BuildStatus.java]: Yes: https://github.com/project-ncl/pnc/blob/master/build-coordinator/src/main/java/org/jboss/pnc/coordinator/notifications/buildTask/BuildStatusNotifications.java#L66 (this is using BuildCoordinatorStatus, so we need an equivalent for BuildStatus)

**Comment by thescouser89** [comment]
@jbartece Unfortunately I won't be able to fix the test failures before the end of the sprint. Sorry about that :(

**Comment by thescouser89** [comment]
retest this please

**Comment by thescouser89** [comment]
retest this please

**Pull request merged**